### PR TITLE
support keyboard binding

### PR DESCRIPTION
**Pull request by hoatle**
- get keyboad binding done
- fix bugs

### CODE REVIEW

File: app/scripts/pixelate.js [hunk 1]
```diff
@@ -244,6 +244,23 @@ var Backbone = Backbone || ({
             masked: true,
             strokeStyle: 'black'
         },
+        keyboard: {//Keyboard binding
+            undo: 90,//ctrl + z
+            redo: 89,//ctrl + y
+            pixelate: 13,//Enter
+            move: {
+                'Top': 38,//Top
```
Comment (hoatle) [inline app/scripts/pixelate.js]: use lower case: `top, left, right, bottom`

**Comment by hoatle** [comment]
- need to test on IE9+
- enable keyboard by default, need to have option to disable it (global and instance options) like:
  `pixelate.keyboardEnabled = true // or false;` as global
  or:

```
pixelate(canvas, {
  keyboardEnabled: true // or false as instance option
});
```

**Comment by hoatle** [comment]
learn https://craig.is/killing/mice

**Comment by tung7ran** [comment]
1. test in IE9+ | passed.
2. enable keyboard | added in pixelate.js.
3. lowcase Top, Right, Bottom, Left | passed.

**Comment by hoatle** [comment]
let's use Mousetrap is if it's available like how we're using _ and Backbone.Events

File: app/scripts/pixelate.js [hunk 2]
```diff
@@ -970,7 +1010,86 @@ var Backbone = Backbone || ({
 
         }
     });
+      //initkeyboard 
+    _.extend(Pixelate.prototype, {
+        initkeyboard: function() {
+            var k = this;
+            this._selectorCanvas.addEventListener('keydown', function (e) {
```
Comment (hoatle) [inline app/scripts/pixelate.js]: should use jQuery instead to make sure it's always cross-browsered.

**Comment by hoatle** [comment]
@codertran use bower to install Mousetrap

**Comment by hoatle** [comment]
@datphan please help reviewing

**Comment by datphan** [comment]
menu_002
- That console error also prevent `pxlImgBySrc()` function on `main.js`.
- Mousetrap is not working.

File: app/scripts/pixelate.js [hunk 3]
```diff
@@ -969,11 +978,38 @@ var Backbone = Backbone || ({
         disposeUISelector: function () {
 
         }
+
     });
+    _.extend(Pixelate.prototype, {
+        initkeyboard: function() {
+            var Mouse = this;
```
Comment (hoatle) [inline app/scripts/pixelate.js]: var self = this;

**Comment by datphan** [comment]
selection_003

`pxlImgBySrc` still not working.

**Comment by mrledon** [comment]
hi all

File: app/scripts/pixelate.js [hunk 4]
```diff
@@ -691,6 +703,21 @@ var Backbone = Backbone || ({
             this.trigger('unmask', se);
             return this;
         },
+        keyboardDisable: function () {
```
Comment (hoatle) [inline app/scripts/pixelate.js]: disableKeyboard

File: app/scripts/pixelate.js [hunk 5]
```diff
@@ -691,6 +703,21 @@ var Backbone = Backbone || ({
...
+            if(!this.isKeyboardEnabled()) {
+                return this;
+            }
+            this._$selectorCanvas.off('keydown').keydown(false);
+            this._keyboardEnabled = false;
+            this.trigger('keyboardDisable');
+            return this;
+        },
+        keyboardEnable: function () {
```
Comment (hoatle) [inline app/scripts/pixelate.js]: enableKeyboard

**Comment by hoatle** [comment]
closed this as no activity.

**Pull request closed without merging**